exec/java-exec/src/main/java/org/apache/drill/exec/vector/complex/fn/JsonReader.java

```diff
@@ -510,10 +517,13 @@ private void writeDataAllText(MapWriter map, FieldSelection selection,
       case VALUE_NUMBER_FLOAT:
       case VALUE_NUMBER_INT:
       case VALUE_STRING:
+        removeNotNullColumn(fieldName);
```

Comment:
I feel uncomfortable since the fix added 'removeNotNullColumn(fieldName)' on the code path that is supposed to be hotspot. Could the reader just clean 'path' if it knows it is recovering from some errors?